src/app/components/tags.ts

```diff
@@ -0,0 +1,36 @@
+import { BaseComponent, type ElementFnProps } from '@components/base-component';
+
+export const span = (props: ElementFnProps<HTMLElement>, ...children: BaseComponent[]) =>
+  new BaseComponent({ ...props, tag: 'span' }, ...children);
+
+export const main = (props: ElementFnProps, ...children: BaseComponent[]) =>
+  new BaseComponent({ ...props, tag: 'main' }, ...children);
+
+export const label = (props: ElementFnProps, ...children: BaseComponent[]) =>
+  new BaseComponent({ ...props, tag: 'label' }, ...children);
+
+export const input = (props: ElementFnProps & Partial<HTMLInputElement>) =>
+  new BaseComponent<HTMLInputElement>({ ...props, tag: 'input' });
+
+export const iconFromCode = (props: ElementFnProps, code: string) =>
+  new BaseComponent({ ...props, tag: 'i', innerHTML: code });
```

Comment:
Function iconFromCode create BaseComponent without specifying a generic type. Consider adding explicit types for clarity.
```suggestion
export const iconFromCode = (props: ElementFnProps,<HTMLElement> code: string) =>
  new BaseComponent<HTMLElement>({ ...props, tag: 'i', innerHTML: code });
```